app/controllers/users/emails_controller.rb

```diff
@@ -62,6 +63,14 @@ def delete
       redirect_to account_url
     end
 
+    def cancel_link_url
+      if session[:sp].present?
```

Comment:
A user can access their account page to add an email even if there's an associated service provider, so I don't think we should use that as the trigger for sending the user to the completions screen.

I could imagine a few other ideas for how we could handle this:

* Look at the referer to see if they came from the completions page, perhaps using [`ParseControllerFromReferer`](https://github.com/18F/identity-idp/blob/main/app/services/parse_controller_from_referer.rb)
* Embed a query parameter in the URL to indicate where they came from
* Handle "Add new email" as a form submission which sets a session value that we use to differentiate where they came from
* Create a separate controller for adding an email from the completions screen

Depending on how involved it is, I might have a soft preference for the last (In my head, I'm imagining it might be simpler than it seems, since the screen is pretty simple and we may be able to reuse some code?). Otherwise, the embedded query parameter might be my second preference.

Reply:
I do recall at refinement you suggested creating a page for this variation of adding an email. I think I'd like to go that route.